quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/QuarkusTextDocumentService.java

```diff
@@ -199,6 +203,33 @@ public void didSave(DidSaveTextDocumentParams params) {
 		});
 	}
 
+	@Override
+	public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormattingParams params) {
+		QuarkusProjectInfoParams projectInfoParams = createProjectInfoParams(params.getTextDocument(), null);
+		return getProjectInfoCache().getQuarkusProjectInfo(projectInfoParams).thenComposeAsync(projectInfo -> {
+			if (projectInfo.getProperties().isEmpty()) {
+				return null;
+			}
+			return getPropertiesModel(params.getTextDocument(), (cancelChecker, document) -> {
+				return getQuarkusLanguageService().doFormat(document, sharedSettings.getFormattingSettings());
+			});
+		});
+	}
+
+	@Override
+	public CompletableFuture<List<? extends TextEdit>> rangeFormatting(DocumentRangeFormattingParams params) {
+		QuarkusProjectInfoParams projectInfoParams = createProjectInfoParams(params.getTextDocument(), null);
```

Comment:
You don't need to get Quarkus properties in the formatting feature, you can remove this code (do like documentSymbol service)